src/datasets/load.py

```diff
@@ -544,7 +545,7 @@ def prepare_packaged_module(name):
                         dataset_info = hf_api.dataset_info(
                             repo_id=path, revision=script_version, token=download_config.use_auth_token
                         )
-                    except Exception:
+                    except requests.exceptions.RequestException:
```

Comment:
Now the test suite is green: all `requests` exceptions (and only them) are catched. I'm wondering if there could be another kind of exception thrown by `HfApi.dataset_info`...

Reply:
Reading https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/hf_api.py#L524, I don't think so. On the other end, as the huggingface_hub does not officially state that the exceptions of this method are requests.exceptions.RequestException, it might break if they touch the code.